test/unit/math/prim/mat/prob/ordered_logistic_test.cpp

```diff
@@ -165,6 +165,16 @@ TEST(ProbDistributions, ordered_logistic) {
   Eigen::Matrix<double, Eigen::Dynamic, 1> lambda_small(3);
   lambda_small << 1, 1, 1;
   EXPECT_THROW(ordered_logistic_log(y, lambda_small, c), std::invalid_argument);
+
+  Eigen::Matrix<double, Eigen::Dynamic, 1> c_small(K - 2);
+  c_small << -0.3, 0.1;
+  std::vector<Eigen::Matrix<double, Eigen::Dynamic, 1>> c_small_vec(4);
+  c_small_vec[0] = c;
+  c_small_vec[1] = c;
+  c_small_vec[2] = c_small;
+  c_small_vec[3] = c;
+  EXPECT_THROW(ordered_logistic_log(y, lambda, c_small_vec),
+               std::invalid_argument);
```

Comment:
I'd like to see a spot check test if the second argument is an int or a std::vector<int>. (Just to make sure integer math doesn't make stuff funky somewhere)

Also, there should be at least some EXPECT_NO_THROW tests that verify vectorization and all combinations thereof. Like, you don't have to check the output on these. Just make sure that the three arguments can each be a vectorized or non-vectorized type (so 8 things in total).

So like:
First argument int or std::vector<int>
Second argument double or Eigen::VectorXd
Third argument Eigen::VectorXd or std::vector<Eigen::VectorXd>

If these combos are covered elsewhere just say that and it's fine. Most of them probly are but I didn't write them down as I looked through.

Reply:
> I'd like to see a spot check test if the second argument is an int or a std::vector. (Just to make sure integer math doesn't make stuff funky somewhere)

As in testing if lambda is an integer?

> Also, there should be at least some EXPECT_NO_THROW tests that verify vectorization and all combinations thereof.

These are already covered, since there aren't many valid combinations (because both ```y``` and ```lambda``` have to have the same length). All possible combinations are instatiated throughout the ```prim/fwd/mix/rev``` testing.

Reply:
> The parser (on the stan side) wouldn't allow that to be called, since it's not one of the valid signatures. Should I add it as a just-in-case?

My bad. Good point.

> These are already covered

Niiice

Reply:
Actually, coming back to this, if someone types ```expression(0)``` in the Stan language, that 0 stays 0 when it's translated to C++, and then might be interpreted as an integer courtesy the templating. So ```expression(0.0)``` might be tested and work but then ```expression(0)``` fails. Happened to me here: https://github.com/stan-dev/math/issues/935, among other times.

Maybe one rev mode test with ints in the lambdas, and then one with ints in the cutpoints? That should have ints touching all the code.

Reply:
We don't have to worry about ints in the cutpoints, since vector_seq_view doesn't accept std::vector<int> arguments and won't compile. For ints in the lambdas I've added an "is_integer" check in the error checking